lib/topological_inventory/sync/inventory_upload_worker.rb

```diff
@@ -18,7 +20,55 @@ def perform(message)
         payload = JSON.parse(message.payload)
         return unless payload["service"] == "topological-inventory"
 
-        logger.info("#{payload}")
+        request = Typhoeus::Request.new(payload["url"])
+        response = request.run
+
+        if response.return_code == :ok # response.success? doesn't work for file:///...
+          data = JSON.parse(response.body)
+
+          source = find_or_create_source(data["name"], data["source"], payload["rh_account"]) unless data["source"].nil?
+
+          if source.present?
+            send_to_ingress_api(source, data, payload["rh_account"])
+          else
+            # TODO: Ack message in kafka when error occurs?
+          end
+        else
+          msg = ["Failed to get uploaded file: "]
+          {
+            :code => response.code,
+            :response_headers => response.headers,
+            :response_body => response.body,
+            :response_status_message => response.status_message
+          }.each_pair do |header, value|
+            msg << "#{header}: #{value}" unless value.blank?
+          end
+          logger.error(msg.join(", "))
+        end
+      end
+
+      private
+
+      def send_to_ingress_api(source, data, external_tenant)
+
+      end
+
+      def find_or_create_source(source_name, source_uid, external_tenant)
+        api_client = sources_api_client(external_tenant)
+
+        sources = api_client.list_sources({:filter => {:uid => source_uid}})
+
+        if sources.data.blank?
+          # TODO: find_or_create_source_type()
```

Comment:
We don't actually want to dynamically create the source type, so we should find the source_type by name first and if it doesn't exist bail out.